Add System Info (Closes #1782)

Citiroller opened the pull request:
This is a initial commit to try and add the system information to lutris as proposed in #1782 
It's not complete as of yet.
When the Preferences are opened on a system level (not on a game level) a new tab appears which contains the system information.
I also changed the `get_mem_info()` function to read from `/proc/meminfo` as the keys were localized in the old function. The old function was deleted. Another possibility to have the keys always in english would be  to export the system locale before running the `free` command with `export LANG=en_US.UTF-8 && free`.

Missing features / issues:
* Only displays information about the first CPU socket, could be problematic with server boards which have two or more CPU sockets.
* Needs testing on different hardware configurations, especially with amdgpu drivers.

![Bildschirmfoto vom 2019-05-22 13-47-29](https://user-images.githubusercontent.com/6241018/58173070-a49bea00-7c9a-11e9-9f23-6c83d2ce9290.png)


As this is my first commit to this project I'm happy about feedback.

strycore commented:
That looks pretty cool so far

jaymehoward commented:
Couple things I'd love to see in this:  Proper column splits with alignment from top to bottom on the page, and a button to get it in a fixed-width font text box, so it can easily be selected and copied for pasting into issues with the insert code option.

isugimpy commented:
Couple things I'd love to see in this:  Proper column formatting from top to bottom for visibility, and button that displays a fixed-width font text box that could be copied and pasted into code tags on issues.

Citiroller commented:
I'll try using the integrated python formatting to format it like two columns.
Regarding the copy function: I use a text box which is non editable, so it's possible to mark the text and copy it by using ctrl+c, I can add a button wich says copy to put the text into the system copy buffer if that is what you propose.

isugimpy commented:
Ah, I missed the fact that it was already a text box.  If that's the case, there's no need for a button!

Citiroller commented:
I just added some graphics information for gpus using mesa. I will add Nvidia graphics info soon. I need to clone my repo to my other machine to test it. I'm not sure if this will work with the amdgpu drivers though as there doesn't seem to be an implementation yet to get the information.

isugimpy commented on lutris/util/linux.py:
```diff
@@ -390,5 +394,66 @@ def gather_system_info():
+    system_dict["OS"] = ' '.join(system_info["dist"])
+    system_dict["Arch"] = system_info["arch"]
+    system_dict["Kernel"] = system_info["kernel"]
+    system_dict["Desktop"] = system_info["env"]["XDG_CURRENT_DESKTOP"]
```
You may want to make this line and the one below it call .get() instead of just doing the subscript lookup.  If somebody's not running a full DE and their WM doesn't present that variable to the env, it'll raise an exception.

Citiroller replied:
So something like
```suggestion
    system_dict["Desktop"] = system_info["env"].get("XDG_CURRENT_DESKTOP", "No DE found")
```

isugimpy replied:
Seems reasonable to me.  Not a maintainer, so I don't want to make a decision on the verbiage, but functionally that's what I was proposing.

Citiroller replied:
Good, I'll wait on more info from the maintainers. Maybe they want me to expose more information or remove some lines. Thanks for your comment!

LiamDawe commented:
If I might have one suggestion: The ability to click to copy to clipboard? Could end up being useful for people to paste their sys info.

strycore commented:
Yeah, the text should totally be selectable

Citiroller commented:
In this implementation the text is displayed in a non editable text box, so the text is selectable and you can copy it by pressing ctrl+c or right click and select copy
![Bildschirmfoto vom 2019-05-26 20-09-41](https://user-images.githubusercontent.com/6241018/58385546-3d5f9c00-7ff2-11e9-8d36-6d172d65ce75.png)
But yeah, I can add a button, for convenience.

LiamDawe commented:
Great, top stuff!

Citiroller commented:
I've added a copy button at the end of the textview. See the attached screenshot.
![Bildschirmfoto vom 2019-05-26 22-26-08](https://user-images.githubusercontent.com/6241018/58386846-01363680-8006-11e9-90d6-11ca0e0dd163.png)
In the last commit I've also added the changes suggested by @isugimpy .
Is there any more information the system info should expose like missing libs? And I hope the way I've implemented it is fine with the maintainers.

Medath commented:
This is a great feature, but for some reason the tabs seem to be a bit messed up on my system
![SCR_20190623_232451](https://user-images.githubusercontent.com/40150792/59982215-055a7180-960f-11e9-9ec2-cacadd2eda05.png)

Citiroller commented:
The thing is, when a non monospace font is used I can't calculate the number of tabs I need for each line to fit perfectly. One way could be to use a monospace font. Or use two text boxes? But I think then you can't select the whole text.
Does anyone know a lib or some function in gtk to do this?
